packages/stardog-language-utils/src/AbstractLanguageServer.ts

```diff
@@ -26,22 +43,70 @@ export abstract class AbstractLanguageServer<
     this.connection.listen();
   }
 
-  handleUninitializedRequest: lsp.StarRequestHandler = () =>
-    new lsp.ResponseError(
-      lsp.ErrorCodes.ServerNotInitialized,
+  handleFoldingRanges(
+    params: FoldingRangeRequestParam,
+    enablePlainPrefix: boolean,
+    enableAtPrefix: boolean
+  ): FoldingRange[] {
+    const { uri } = params.textDocument;
+    const document = this.documents.get(uri);
+    const ranges: FoldingRange[] = [];
+
+    if (!document) {
+      return ranges;
+    }
+
+    const lines = document.getText().split(/\r?\n/);
+    let lineIdx = 0;
+    while (lineIdx < lines.length - 1) {
+      const lowerCaseLine = lines[lineIdx].toLowerCase();
+      const lowerCaseNextLine = lines[lineIdx + 1].toLowerCase();
+      const trimmedLine = lowerCaseLine.trimLeft();
+      const trimmedNextLine = lowerCaseNextLine.trimLeft();
+      const indentLevel = lowerCaseLine.length - trimmedLine.length;
+      const indentNextLevel = lowerCaseNextLine.length - trimmedNextLine.length;
+      if (
+        enablePlainPrefix &&
+        trimmedLine.startsWith('prefix') &&
+        trimmedNextLine.startsWith('prefix')
+      ) {
+        const range = this.getPrefixFoldingRange(lines, lineIdx);
+        if (range) {
+          ranges.push(range);
+          lineIdx = range.endLine;
+        } else {
+          lineIdx++;
+        }
+      } else if (
+        enableAtPrefix &&
+        trimmedLine.startsWith('@prefix') &&
+        trimmedNextLine.startsWith('@prefix')
+      ) {
+      } else if (trimmedLine && indentNextLevel > indentLevel) {
```

Comment:
Is the empty pair of braces here intentional? Seems like something's missing, but I'm not sure what's going on here and would like a comment if it is intentional.

Reply:
Brain fart, think i pasted it down there and forgot. but i think it's cleaner (less code) to just leave it as it was before